src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -200,27 +199,17 @@ Assembly TryLoadAssembly(AssemblyName assemblyName)
                 // Assembly resolution is not thread-safe.
                 lock (loadedAssemblies)
                 {
-                    Assembly assembly;
-                    if (loadedAssemblies.TryGetValue(assemblyName.FullName, out assembly))
+                    if (loadedAssemblies.TryGetValue(assemblyName.FullName, out Assembly assembly))
                     {
                         return assembly;
                     }
 
-                    if (IsMSBuildAssembly(assemblyName))
+                    string targetAssembly = Path.Combine(msbuildPath, assemblyName.Name + ".dll");
```

Comment:
```suggestion
                    // Look in the MSBuild folder for any unresolved reference--it may be a dependency
                    // of MSBuild or a task.
                    string targetAssembly = Path.Combine(msbuildPath, assemblyName.Name + ".dll");
```